packages/query/src/rate-limiter/best-in-slot-limiter.ts

```diff
@@ -0,0 +1,36 @@
+import PQueue from 'p-queue';
+
+import { BESTINSLOT_API_BASE_URL_TESTNET } from '@leather.io/models';
+
+import { useCurrentNetworkState } from '../leather-query-provider';
+import { PriorityQueue } from './queue-class';
+
+const bestInSlotMainnetApiLimiter = new PQueue({
+  interval: 1000,
+  intervalCap: 4,
+  timeout: 60000,
+  queueClass: PriorityQueue,
+});
+
+const bestInSlotTestnetApiLimiter = new PQueue({
+  concurrency: 20,
+  interval: 60000,
+  intervalCap: 20,
+  timeout: 60000,
+});
+
+export function useBestInSlotApiRateLimiter(): PQueue {
+  const currentNetwork = useCurrentNetworkState();
+
+  switch (currentNetwork.mode) {
+    case 'mainnet':
+      return bestInSlotMainnetApiLimiter;
+    case 'testnet':
```

Comment:
Maybe we should add a `default` here too for `bestInSlotTestnetApiLimiter` in case its a custom network?

Reply:
network mode could be only `mainnet` or `testnet`. we have helper fn for such cases `bitcoinNetworkModeToCoreNetworkMode`